### ISSUE TITLE
FPS cap with V-Sync enabled is incorrect with multiple monitors with different refresh rates

### ISSUE THREAD

**Issue by bogcow** [godotengine/godot#89681]
### Tested versions

- Reproducible in 4.2.stable
- Not reproducible in 3.5.3.stable

### System information

Windows 10

### Issue description

I have two monitors: one is 144Hz and the other is 60Hz. If I run an empty project with V-Sync enabled, the FPS is always capped at 144. Even if I drag the game window to my 60Hz monitor, the game is still capped at 144 FPS. This is different from Godot 3.5.3, where if I dragged the game window to my 60Hz monitor, the FPS cap would update to 60, which I believe should be the intended behavior.
Even if I launch the game window on my 60Hz monitor by setting the initial_position_type in ProjectSettings, the FPS is still capped at 144 rather than 60.

### Steps to reproduce

Create a new project with V-Sync enabled. Check the fps as you move the game window between two monitors with different refresh rates.

### Minimal reproduction project (MRP)

N/A

**Comment by AThousandShips** [comment]
Can you test this on 4.2.1 or [4.2.2.rc2](https://godotengine.org/article/release-candidate-godot-4-1-4-and-4-2-2-rc-2/), or even better on [4.2.dev5](https://godotengine.org/article/dev-snapshot-godot-4-3-dev-5/)? I know some issues with frame rate has been solved not too long ago

**Comment by bogcow** [comment]
I just tried testing on 4.2.2.rc2 and 4.2.dev5 and the issue is present on both versions as well

**Comment by brcontainer** [comment]
> Can you test this on 4.2.1 or [4.2.2.rc2](https://godotengine.org/article/release-candidate-godot-4-1-4-and-4-2-2-rc-2/), or even better on [4.2.dev5](https://godotengine.org/article/dev-snapshot-godot-4-3-dev-5/)? I know some issues with frame rate has been solved not too long ago

Do you mean 4.3dev5?

---

@bogcow Did you test on 4.3? Download: https://godotengine.org/download/archive/

**Comment by bogcow** [comment]
Oh, I meant to say 4.3.dev5, yes the issue is present on 4.3.dev5

**Comment by almoandrioli** [comment]
I have a similar problem but in my case the fps is not capped even if vsync is on and Nvidia Control Panel has fps limit on.

**Comment by Calinou** [comment]
> I have a similar problem but in my case the fps is not capped even if vsync is on and Nvidia Control Panel has fps limit on.

Does FPS get capped with [RTSS](https://www.guru3d.com/download/rtss-rivatuner-statistics-server-download/)' framerate limiter active?

**Comment by almoandrioli** [comment]
> > I have a similar problem but in my case the fps is not capped even if vsync is on and Nvidia Control Panel has fps limit on.
> 
> Does FPS get capped with [RTSS](https://www.guru3d.com/download/rtss-rivatuner-statistics-server-download/)' framerate limiter active? This is more reliable than NVIDIA's method in my experience, and it can be adjusted while a game is already running.

No, even with msi afterburner frame limiter the 3d editor fps is not capped. But msi afterburner overlay shows 60 fps lock but godot editor show a much higher fps (500~)

**Comment by VertexMachine** [comment]
<img width="422" height="175" alt="Image" src="https://github.com/user-attachments/assets/c137688d-c78d-40af-a990-a8906bab5b48" />

I'm not sure if this is helpful or not, but seems that Engine.GetFramesPerSecond gets confused on multi-monitor setup as well. I have 3 monitors (180Hz, 144Hz and 60Hz with 144Hz being the main one). FrameView shows similar fps to Engine.GetFramesPerSecond on both 144Hz and 180Hz monitors (i.e., with vsync the 180Hz is capped at 144 fps as the main monitor). But on that 60Hz monitor somehow vsync did kick in and frames are capped to ~60fps as shown by FrameView, but Engine.GetFramesPerSecond shows incorrect 144fps.

Edit, for completeness. Same on main display:

<img width="577" height="233" alt="Image" src="https://github.com/user-attachments/assets/8ff1f1f4-23f9-48bf-bbee-d507446d781c" />

And  on 180Hz one:

<img width="427" height="201" alt="Image" src="https://github.com/user-attachments/assets/ab9152a0-4a0c-4f93-9a4f-9b72cab9c646" />

the code for displaying stuff on the title is simply that:
```csharp
    public override void _Process(double delta)
    {
        int screen = DisplayServer.WindowGetCurrentScreen();
        float refreshRate = DisplayServer.ScreenGetRefreshRate(screen);
        var vsync = DisplayServer.WindowGetVsyncMode();


        DisplayServer.WindowSetTitle(
            $"{_gameTitle} | FPS: {Engine.GetFramesPerSecond():F0}" +
            $" | Screen: {screen}" +
            $" | Hz: {refreshRate:F1}" +
            $" | VSync: {vsync}" +
            $" | Max FPS: {Engine.MaxFps}");
    }
```

(this is from a running game, not the editor FPS display; On: v4.7.2.stable.mono.official [ed1daf0bf])
